src/server/mod.rs

```diff
@@ -299,29 +322,29 @@ where A: Accept,
 
 /// A handle of the running server.
 pub struct Listening {
-    addr: SocketAddr,
+    addrs: Vec<SocketAddr>,
     shutdown: (Arc<AtomicBool>, rotor::Notifier),
 }
 
 impl fmt::Debug for Listening {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
         f.debug_struct("Listening")
-            .field("addr", &self.addr)
+            .field("addrs", &self.addrs)
             .field("closed", &self.shutdown.0.load(Ordering::Relaxed))
             .finish()
     }
 }
 
 impl fmt::Display for Listening {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        fmt::Display::fmt(&self.addr, f)
+        fmt::Debug::fmt(&self.addrs, f)
```

Comment:
Hm, before this would allow a `Listening` to be printed nicely, by doing `println!("listening on {}", listening)`. This change will make the output look like `listening on [V4(127.0.0.1:8080)]`.

I suppose if there are muliple addrs, it likely should output a comma separated list. But either way, it'd probably be best to not have the `Debug` output of the `SocketAddr`s.